scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -176,6 +180,54 @@ object BeamOp extends Serializable {
     }
   }
 
+  final case class HashJoinOp[K, V, U, W](
+      left: BeamOp[(K, V)],
+      right: BeamOp[(K, U)],
+      joiner: (K, V, Iterable[U]) => Iterator[W]
+  )(implicit kryoCoder: KryoCoder, ordK: Ordering[K])
+      extends BeamOp[(K, W)] {
+    override def run(pipeline: Pipeline): PCollection[_ <: (K, W)] = {
+      val leftPCollection = left.run(pipeline)
+      val keyCoder: Coder[K] = OrderedSerializationCoder.apply(ordK, kryoCoder)
+      val rightPCollection: PCollection[(K, U)] = widenPCollection(right.run(pipeline))
+
+      val tuple: PCollectionTuple = PCollectionTuple.of[(K, _)](
+        "left",
+        widenPCollection(leftPCollection): PCollection[(K, _)],
+        "right",
+        widenPCollection(rightPCollection): PCollection[(K, _)]
+      )
+
+      tuple.apply(HashJoinTransform(keyCoder, joiner))
```

Comment:
Yes that is true but for all the `apply` from Beam's Pipeline, PTransforms and similar we write it explicitly to signal it's Beam API calls and that's what it would look like called from Java or from Beam examples.